v2/src/datadog.ts

```diff
@@ -32,28 +32,27 @@ import {
 
 const versionJson = require("../version.json");
 
-type IDatadogPropsV2 = IDatadogProps & { apiKeySecret?: ISecret };
-
-class Datadog extends Construct {
+export class Datadog extends Construct {
   scope: Construct;
-  props: IDatadogPropsV2;
+  props: DatadogPropsV2;
   transport: Transport;
-  constructor(scope: Construct, id: string, props: IDatadogPropsV2) {
+  constructor(scope: Construct, id: string, props: DatadogPropsV2) {
     if (process.env.DD_CONSTRUCT_DEBUG_LOGS?.toLowerCase() == "true") {
       log.setLevel("debug");
     }
     super(scope, id);
     this.scope = scope;
     this.props = props;
+    let apiKeySecretArn = this.props.apiKeySecretArn;
     if (this.props.apiKeySecret !== undefined) {
-      this.props.apiKeySecretArn = this.props.apiKeySecret.secretArn;
+      apiKeySecretArn = this.props.apiKeySecret.secretArn;
```

Comment:
declaring `DatadogPropsV2 ` as a behavioral interface is preferred as we can have non-readonly values, so that we can directly mutate apiKeySecretArn.

This isn't possible because it changes the way the datadog construct is declared, thus making it non-backwards compatable. See below:

before:
```
datadog = Datadog(
            self,
            "Datadog",
            extension_layer_version=43,
            python_layer_version=72,
            site="datadoghq.com",
            api_key_secret=secret,
        )
```

non-backwards compatible change introduced IF we use behavioral interfaces:
```
datadog = Datadog(
            self,
            "Datadog",
            {
                extension_layer_version=43,
                python_layer_version=72,
                site="datadoghq.com",
                api_key_secret=secret,
            }
        )
```

therefore we can't use behavioral interfaces, and have to live with passing a apiKeySecretArn override flag as a parameter to `validateProps `